### PR TITLE
NBL-2: CPP: Demo for Cluster API

### PR DESCRIPTION
**Pull request by FedorUporov**

### CODE REVIEW

File: modules/platforms/cpp/examples/cluster-compute-example/src/cluster_compute_example.cpp [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        Ignite client = Ignition::Start(cfgs[3], "Client");
+
+        std::cout << std::endl;
+        std::cout << ">>> Cluster compute example started." << std::endl;
+        std::cout << std::endl;
+
+        // Get binding instances and register our classes as a compute functions.
+        node1.GetBinding().RegisterComputeFunc<PrintMsg>();
+        node2.GetBinding().RegisterComputeFunc<PrintMsg>();
+        node3.GetBinding().RegisterComputeFunc<PrintMsg>();
```
Comment (isapego) [inline modules/platforms/cpp/examples/cluster-compute-example/src/cluster_compute_example.cpp]: I believe, this function should be registered for the client node as well.

File: modules/platforms/cpp/examples/cluster-compute-example/src/cluster_compute_example.cpp [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+        // Create cluster groups splitted up by demo attribute value.
+        ClusterGroup localGroup = client.GetCluster().ForAll();
+        ClusterGroup group1 = localGroup.ForAttribute("DemoAttribute", "Value0");
+        ClusterGroup group2 = localGroup.ForAttribute("DemoAttribute", "Value1");
+
+        // Making calls.
+        client.GetCompute(group1).Broadcast(PrintMsg("DemoAttribute=Value0"));
+        client.GetCompute(group2).Broadcast(PrintMsg("DemoAttribute=Value1"));
+
```
Comment (isapego) [inline modules/platforms/cpp/examples/cluster-compute-example/src/cluster_compute_example.cpp]: Insert `std::cin.get()` here, to allow user wait for all compute jobs to execute before shutting down a cluster.

File: modules/platforms/cpp/examples/cluster-compute-example/src/cluster_compute_example.cpp (hunk 1 above)
Comment (FedorUporov) [reply modules/platforms/cpp/examples/cluster-compute-example/src/cluster_compute_example.cpp]: Added.

File: modules/platforms/cpp/examples/cluster-compute-example/src/cluster_compute_example.cpp (hunk 2 above)
Comment (FedorUporov) [reply modules/platforms/cpp/examples/cluster-compute-example/src/cluster_compute_example.cpp]: Added additional user message, if I got your idea correctly.

**Pull request closed without merging**